js/adapt-contrib-accordion.js

```diff
@@ -34,33 +36,65 @@ define(function(require) {
 
         toggleItem: function(event) {
             event.preventDefault();
-            this.$('.accordion-item-body').stop(true, true).slideUp(200);
 
-            if (!$(event.currentTarget).hasClass('selected')) {
-                this.$('.accordion-item-title').removeClass('selected');
-                var body = $(event.currentTarget).addClass('selected visited').siblings('.accordion-item-body').slideToggle(200, function() {
-                  $(body).a11y_focus();
-                });
-                this.$('.accordion-item-title-icon').removeClass('icon-minus').addClass('icon-plus');
-                $('.accordion-item-title-icon', event.currentTarget).removeClass('icon-plus').addClass('icon-minus');
+            var toggleButton = $(event.currentTarget);
+            var accordionItem = toggleButton.parent('.accordion-item');
+            var isCurrentlyExpanded = toggleButton.hasClass('selected');
 
-                if ($(event.currentTarget).hasClass('accordion-item')) {
-                    this.setVisited($(event.currentTarget).index());
-                } else {
-                    this.setVisited($(event.currentTarget).parent('.accordion-item').index());
+            if (this.model.get('_shouldCollapseItems')) {
+                // Close and reset all Accordion items
+                var allAccordionItems = this.$('.accordion-item');
+                var count = allAccordionItems.length;
+                for (var i = 0; i < count; i++) {
+                    this.closeItem($(allAccordionItems[i]));
                 }
             } else {
-                this.$('.accordion-item-title').removeClass('selected');
-                $(event.currentTarget).removeClass('selected');
-                $('.accordion-item-title-icon', event.currentTarget).removeClass('icon-minus').addClass('icon-plus');
+                // Close and reset the selected Accordion item
+                this.closeItem(accordionItem);
             }
-            // set aria-expanded value
-            if ($(event.currentTarget).hasClass('selected')) {
-                $('.accordion-item-title').attr('aria-expanded', false);
-                $(event.currentTarget).attr('aria-expanded', true);
-            } else {
-                $(event.currentTarget).attr('aria-expanded', false);
+
+            if (!isCurrentlyExpanded) {
+                this.openItem(accordionItem);
+            }
+        },
+
+        closeItem: function(itemEl) {
+            if (!itemEl) {
+                return false;
+            }
+
+            var body = $('.accordion-item-body', $(itemEl)).first();
```

Comment:
I think I'm right in saying that `itemEl` is a JQuery object? If so, there's no need to do this:
```
var body = $('.accordion-item-body', $(itemEl)).first();
```
Cause you can just do this: 
```
var body = $('.accordion-item-body', itemEl).first();
```
It should really be renamed `$itemEl` to make it clear that it is a JQuery object

Again, this doesn't actually affect the functionality at all so you can leave as-is if you want, I'm aware that you've already had to make a lot of changes to get this PR this far - sorry!!